plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -506,10 +506,15 @@ private TestExecutionResult testPluginAgainst(MavenCoordinates coreCoordinates,
                             String dependencies = manifest.getMainAttributes().getValue("Plugin-Dependencies");
                             if (dependencies != null) {
                                 // e.g. matrix-auth:1.0.2;resolution:=optional,credentials:1.8.3;resolution:=optional
-                                for (String pair : dependencies.replace(";resolution:=optional", "").split(",")) {
-                                    String[] nameVer = pair.split(":");
+                                for (String pair : dependencies.split(",")) {
+                                    boolean optional = false;
+                                    if (pair.split(";").length > 1) {
```

Comment:
This seems unnecessarily fragile in case there's ever another key-value pair possible after `;`.